zebra-rpc/src/server.rs

```diff
@@ -22,9 +23,14 @@ pub mod compatibility;
 #[derive(Clone, Debug)]
 pub struct RpcServer;
 
+lazy_static::lazy_static! {
+    /// Storage for the application version string coming from zebrad.
+    pub static ref APP_VERSION: Mutex<Vec<u8>> = Mutex::new(vec![]);
+}
+
 impl RpcServer {
     /// Start a new RPC server endpoint
-    pub fn spawn(config: Config) -> tokio::task::JoinHandle<()> {
+    pub fn spawn(config: Config, app_version: String) -> tokio::task::JoinHandle<()> {
         if let Some(listen_addr) = config.listen_addr {
             info!("Trying to open RPC endpoint at {}...", listen_addr,);
 
```

Comment:
Suggestion (2/3): initialise the the RPC method object with the string.

You'll also need to pass `rpc_impl` to `extend_with`. (GitHub won't let me do suggestions on that line.)

```suggestion

let rpc_impl = RpcImpl { app_version };

```

Reply:
Thanks, this is the part i had missing when i did it this way. I refactored everything at https://github.com/ZcashFoundation/zebra/pull/3660/commits/dd0b18fa18c20e220c9981b12f5ebc65cafd7859